internal/controller/certificates/policies/policies.go

```diff
@@ -96,14 +96,16 @@ func NewReadinessPolicyChain(c clock.Clock) Chain {
 // correctness of metadata and output formats of Certificate's Secrets.
 func NewSecretPostIssuancePolicyChain(ownerRefEnabled bool, fieldManager string) Chain {
 	return Chain{
+		SecretBaseLabelsMismatch,
+		SecretCertificateDetailsAnnotationsMismatch,
+		SecretManagedLabelsAndAnnotationsManagedFieldsMismatch(fieldManager),
 		SecretTemplateMismatchesSecret,
 		SecretTemplateMismatchesSecretManagedFields(fieldManager),
```

Comment:
I can't understand why there are 5 separate functions all comparing the labels and annotations of the Secret resource.
Don't they all do basically the same checks?
Add some comments here explaining what all these checks are for... OR add TODO comments suggesting how these functions could be consolidated.

Reply:
done